### PR TITLE
[MINOR] remove datafusion-cli's ballista feature from docs

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?
Now we not support connect ballista using datafusion-cli
```
datafusion-cli % cargo build --features ballista
error: none of the selected packages contains these features: ballista

```
It moved to https://github.com/apache/arrow-ballista/tree/master/ballista-cli

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

# Does this PR break compatibility with Ballista?

### CODE REVIEW

**Review by andygrove — APPROVED** [review]

File: docs/source/user-guide/cli.md [hunk 1]
```diff
@@ -19,8 +19,7 @@
 
 # DataFusion Command-line Interface
 
-The DataFusion CLI allows SQL queries to be executed by an in-process DataFusion context, or by a distributed
-Ballista context.
+The DataFusion CLI allows SQL queries to be executed by an in-process DataFusion context.
 
 ```
 USAGE:
```
Comment (waynexia) [inline docs/source/user-guide/cli.md:25]: According to https://github.com/apache/arrow-datafusion/pull/2604/ I think this entire block needs to be updated.

Comment (Ted-Jiang) [reply docs/source/user-guide/cli.md:25]: i think it still need diff between `FLAGS ` and `OPTIONS `

**Review by alamb — APPROVED** [review]
Thanks @Ted-Jiang

**Comment by liukun4515** [comment]
@Ted-Jiang 
From this page https://github.com/apache/arrow-ballista/tree/master/ballista-cli, the document is not fit for the current project, could you please update the doc in the ballista project.
Thanks

**Pull request merged**